### PR TITLE
Implement SQLite helper functions for GDT refactor

### PR DESCRIPTION
**Pull request by mikehaney24**

### CODE REVIEW

File: GoogleDataTransport/GDTCORLibrary/Public/GDTCORSqlite.h [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+/** Convenience typedef to define a block ran on each row of a query result. */
+typedef void (^GDTCORSqliteRowResultBlock)(void);
+
+/** Instantiates a sqlite3 object given the filename.
+ *
+ * @param db The db reference to instantiate.
+ * @param fileURL The file URL to make a db at.
+ * @return YES if the db was successfully opened, NO otherwise.
+ */
+BOOL GDTCORSQLOpenDB(sqlite3 *_Nullable *_Nullable db, NSURL *fileURL);
```
Comment (chliangGoogle) [inline GoogleDataTransport/GDTCORLibrary/Public/GDTCORSqlite.h]: I don't see any of these methods are used. Is this just adding the helper functions so far?

File: GoogleDataTransport/GDTCORLibrary/GDTCORSqlite.m [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+    return NO;
+  }
+  return sqlite3_open([fileURL path].UTF8String, db) == SQLITE_OK;
+}
+
+BOOL GDTCORSQLCloseDB(sqlite3 *db) {
+  return sqlite3_close(db) == SQLITE_OK;
+}
+
+BOOL GDTCORSQLCompileSQL(sqlite3_stmt **stmt, sqlite3 *db, NSString *statement) {
```
[the remark this answers is not in the document]
Comment (mikehaney24) [reply GoogleDataTransport/GDTCORLibrary/GDTCORSqlite.m]: So you think using a const char * might be better than NSString (or even a const NSString) here?

[the remark this answers is not in the document]
Comment (chliangGoogle) [reply GoogleDataTransport/GDTCORLibrary/GDTCORSqlite.m]: I mean like the SQL statement itself (e.g. @"UPDATE %@ SET namespace = ? WHERE namespace = ?") should be defined inside the function, but not passing in.
This is why RCNConfigDBManager is so big cuz we couldn't extract the function out.
This seems like the trend inside g3, and the reason was it is more secure so people won't inject their own sql cmd.

[the remark this answers is not in the document]
Comment (mikehaney24) [reply GoogleDataTransport/GDTCORLibrary/GDTCORSqlite.m]: Maybe I'm not understanding, but I'm not sure that makes sense for this wrapper around sqlite. GDTStorage will eventually have statically defined SQL strings that eventually get routed to these helper functions, but I don't think there's a reasonable way for me to write a sqlite wrapper in such a way that every callable SQL string is pre-programmed in here. I'll think about that.

In any case, I'm not generally concerned about that--I myself am using a third-party program that opens the sqlite file and does stuff to it while my tests are running just so I can validate things. If someone wanted to run sql on this they could just open their own handle to the file and execute arbitrary sql (or copy the sql file, operate on it, and copy it back overwriting the original).

[the remark this answers is not in the document]
Comment (chliangGoogle) [reply GoogleDataTransport/GDTCORLibrary/GDTCORSqlite.m]: I think what you are doing is fine, and I agree it's probably impossible to create wrapper if we are going in that route. Maybe it make more sense to close source and I definitely like the wrapper as it simplifies the code a lot.

File: GoogleDataTransport/GDTCORLibrary/Public/GDTCORSqlite.h (hunk 1 above)
Comment (mikehaney24) [reply GoogleDataTransport/GDTCORLibrary/Public/GDTCORSqlite.h]: Yeah, these are just helper functions. They're only used in tests right now.

**Comment by mikehaney24** [comment]
> LGTM, but please wait to merge until after release-6.11.0 is cut.

It's not being merged to master, so that shouldn't matter in theory. I'm merging all PRs to a feature branch, because this will be a very large change when it's all done.

**Comment by paulb777** [comment]
Oh, I missed that. Please disregard. :P

File: GoogleDataTransport/GDTCORLibrary/GDTCORSqlite.m [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+}
+
+#pragma mark - Public functions
+
+BOOL GDTCORSQLOpenDB(sqlite3 **db, NSURL *fileURL) {
+  if (fileURL == nil || fileURL.path.length == 0) {
+    GDTCORLogError(GDTCORMCEDatabaseError, @"%@", @"A filename for the sqlite db must not be nil");
+    return NO;
+  }
+  return sqlite3_open([fileURL path].UTF8String, db) == SQLITE_OK;
```
[the remark this answers is not in the document]
Comment (mikehaney24) [reply GoogleDataTransport/GDTCORLibrary/GDTCORSqlite.m]: Done.

**Pull request merged**